tests/conftest.py

```diff
@@ -54,16 +59,127 @@ def start_postgresql_container(self):
             name=self.config.CONTAINER_NAME,
             environment=self.db_environment,
             ports=self.db_ports)
+    
+    def stop_if_running(self):
```

Comment:
This looks good. Thank you for building a pattern for it: I can think of a couple of repos that probably need to appropriate your solution.